io/net/infiniband/dapl.py

```diff
@@ -0,0 +1,163 @@
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Manvanthara B Puttashankar <manvanth@linux.vnet.ibm.com>
+
+"""
+dapl test
+"""
+
+import time
+import netifaces
+from netifaces import AF_INET
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process, distro
+from avocado.utils.network.interfaces import NetworkInterface
+from avocado.utils.network.hosts import LocalHost, RemoteHost
+from avocado.utils.ssh import Session
+
+
+class dapl(Test):
+    """
+    dapltest.
+    """
+
+    def setUp(self):
+        """
+        Setup and install dependencies for the test.
+        """
+        self.test_name = "dapltest"
+        self.hprm = self.params.get("HOST_PARAM", default="None")
+        self.pprm = self.params.get("PEER_PARAM", default="None")
+        if self.hprm == "None" and self.pprm == "None":
+            self.cancel("No PARAM given")
+        if process.system("ibstat", shell=True, ignore_status=True) != 0:
+            self.cancel("MOFED is not installed. Skipping")
+        detected_distro = distro.detect()
+        pkgs = []
+        smm = SoftwareManager()
+        if detected_distro.name == "Ubuntu":
+            pkgs.extend(["openssh-client", "iputils-ping"])
+        elif detected_distro.name == "SuSE":
+            pkgs.extend(["openssh", "iputils"])
+        else:
+            pkgs.extend(["openssh-clients", "iputils"])
+        for pkg in pkgs:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("Not able to install %s" % pkg)
+        interfaces = netifaces.interfaces()
+        self.dpl_int = self.params.get("dapl_interface", default="")
+        self.dpl_peer = self.params.get("dapl_peer_interface", default="")
+        self.hprm = self.hprm.replace('$dapl_interface', self.dpl_int)
+        self.pprm = self.pprm.replace('$dapl_peer_interface', self.dpl_peer)
+        self.iface = self.params.get("interface", default="")
+        self.peer_ip = self.params.get("peer_ip", default="")
+        self.hprm = self.hprm.replace('$peer_ip', self.peer_ip)
+        self.peer_user = self.params.get("peer_user", default="root")
+        self.peer_password = self.params.get("peer_password", '*',
+                                             default="None")
+        self.ipaddr = self.params.get("host_ip", default="")
+        self.netmask = self.params.get("netmask", default="")
+        local = LocalHost()
+        self.timeout = "2m"
+        self.session = Session(self.peer_ip, user=self.peer_user,
+                               password=self.peer_password)
```

Comment:
@manvanthar here add self.session.connect() to check the connection to peer or it will not ask password during command run on peer machine and we no need to do ssh-keygen for adding public ip of 
peer.

Reply:
addressed in new patch